ui/StatusQ/src/StatusQ/Popups/StatusMenu.qml

```diff
@@ -85,15 +88,28 @@ Menu {
         }
     }
 
+    QtObject {
+        id: d
+        //helper property to get the max implicit width of the delegate
+        property real maxDelegateImplWidth: 0
+        property real maxDelegateImplHeight: 0
+    }
+
     delegate: StatusMenuItem {
         visible: root.hideDisabledItems ? enabled : true
         height: visible ? implicitHeight : 0
+        onImplicitWidthChanged: {
+            d.maxDelegateImplWidth = Math.max(d.maxDelegateImplWidth, implicitWidth)
+        }
+        onImplicitHeightChanged: {
+            d.maxDelegateImplHeight = Math.max(d.maxDelegateImplHeight, implicitHeight)
+        }
     }
 
     contentItem: StatusListView {
         currentIndex: root.currentIndex
-        implicitHeight: contentHeight
-        implicitWidth: contentWidth
+        implicitHeight: orientation === ListView.Horizontal ? d.maxDelegateImplHeight : contentHeight
```

Comment:
the `orientation` of the list is not exposed directly from the `StatusMenu` so imo it's not necessary to assume that sb can change that (that list view is internal impl detail). So that check can be removed or orientation exposed as an alias